types/wordpress__block-editor/components/block-controls.d.ts

```diff
@@ -1,6 +1,10 @@
 import { Slot, ToolbarGroup } from "@wordpress/components";
 import { FC, JSX, ReactNode } from "react";
 
+type GetArrayTypeFromPossibleNestedArray<TestType extends Record<string,unknown> | Array<Record<string,unknown>>> = TestType extends Array<Record<string,unknown>> ? TestType[number] : TestType;
+
+type BlockControlControlsType = GetArrayTypeFromPossibleNestedArray<NonNullable<Parameters<typeof ToolbarGroup>[0]["controls"]>[number]>;
```

Comment:
@rafaucau here's my fix for the ToolbarGroup array bug

Reply:
Ok, thanks!